### ISSUE TITLE
New ASAN error

### ISSUE THREAD

**Issue by claudioandre** [magnumripper/JohnTheRipper#2573]
FYI
* happens on gcc and clang (in Trusty and Ubuntu "Hot")

```
==16393== ERROR: AddressSanitizer: global-buffer-overflow on address 0x000000aca700 at pc 0x456d69 bp 0x7fffb58f8740 sp 0x7fffb58f8738
READ of size 8 at 0x000000aca700 thread T0
    #0 0x456d68 in KeccakF1600_StateXORBytesInLane /home/travis/build/claudioandre/JohnTheRipper/src/KeccakF-1600-opt64.c:79
    #1 0x4610f8 in Keccak_SpongeAbsorb /home/travis/build/claudioandre/JohnTheRipper/src/KeccakSponge.c:93
    #2 0x460b9a in Keccak_HashUpdate /home/travis/build/claudioandre/JohnTheRipper/src/KeccakHash.c:40
    #3 0x69b3b3 in crypt_all._omp_fn.0 /home/travis/build/claudioandre/JohnTheRipper/src/ethereum_fmt_plug.c:194
    #4 0x69bb74 in crypt_all /home/travis/build/claudioandre/JohnTheRipper/src/ethereum_fmt_plug.c:125
    #5 0x8d6d19 in is_key_right /home/travis/build/claudioandre/JohnTheRipper/src/formats.c:207
    #6 0x8d7d1a in test_fmt_8_bit /home/travis/build/claudioandre/JohnTheRipper/src/formats.c:1165
    #7 0x8db722 in fmt_self_test_body /home/travis/build/claudioandre/JohnTheRipper/src/formats.c:879
    #8 0x8dd55a in fmt_self_test /home/travis/build/claudioandre/JohnTheRipper/src/formats.c:1671
    #9 0x8bc4dd in benchmark_format /home/travis/build/claudioandre/JohnTheRipper/src/bench.c:367
    #10 0x8be311 in benchmark_all /home/travis/build/claudioandre/JohnTheRipper/src/bench.c:809
    #11 0x8eb9d7 in john_run /home/travis/build/claudioandre/JohnTheRipper/src/john.c:1603
    #12 0x8ed256 in main /home/travis/build/claudioandre/JohnTheRipper/src/john.c:2009
    #13 0x7f9e74014f44 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21f44)
    #14 0x406a72 in _start (/home/travis/build/claudioandre/JohnTheRipper/run/john+0x406a72)
0x000000aca702 is located 0 bytes to the right of global variable '*.LC7 (ethereum_fmt_plug.c)' (0xaca700) of size 2
  '*.LC7 (ethereum_fmt_plug.c)' is ascii string ''
SUMMARY: AddressSanitizer: global-buffer-overflow /home/travis/build/claudioandre/JohnTheRipper/src/KeccakF-1600-opt64.c:79 KeccakF1600_StateXORBytesInLane
Shadow bytes around the buggy address:
  0x000080151490: f9 f9 f9 f9 00 06 f9 f9 f9 f9 f9 f9 00 00 00 00
  0x0000801514a0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0000801514b0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0000801514c0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0000801514d0: 00 00 00 00 00 00 00 00 04 f9 f9 f9 f9 f9 f9 f9
=>0x0000801514e0:[02]f9 f9 f9 f9 f9 f9 f9 00 01 f9 f9 f9 f9 f9 f9
  0x0000801514f0: 00 00 f9 f9 f9 f9 f9 f9 00 00 00 00 00 04 f9 f9
  0x000080151500: f9 f9 f9 f9 01 f9 f9 f9 f9 f9 f9 f9 00 00 f9 f9
  0x000080151510: f9 f9 f9 f9 00 03 f9 f9 f9 f9 f9 f9 00 00 00 00
  0x000080151520: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x000080151530: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:     fa
  Heap righ redzone:     fb
  Freed Heap region:     fd
  Stack left redzone:    f1
  Stack mid redzone:     f2
  Stack right redzone:   f3
  Stack partial redzone: f4
  Stack after return:    f5
  Stack use after scope: f8
  Global redzone:        f9
  Global init order:     f6
  Poisoned by user:      f7
  ASan internal:         fe
==16393== ABORTING
```

I'm kind of disconnected. But I guess you guys can easily reproduce the error.

**Comment by kholia** [comment]
This is pretty harmless but I will try to fix it ASAP.

**Comment by kholia** [comment]
Fixed in commit 35ae8db8edf46eb40b40d47671ce5fb106f3279e. Thanks for reporting this problem 👍
